feature-libs/checkout/base/components/components/place-order/place-order.component.spec.ts

```diff
@@ -0,0 +1,147 @@
+  launch(
+    _caller: LAUNCH_CALLER | string,
+    _vcr?: ViewContainerRef,
+    _data?: any
+  ): void | Observable<ComponentRef<any> | undefined> {}
+  clear(_caller: LAUNCH_CALLER | string): void {}
+}
+
+@Pipe({
+  name: 'cxUrl',
+})
+class MockUrlPipe implements PipeTransform {
+  transform(): any {}
+}
+
+describe('PlaceOrderComponent', () => {
+  let component: PlaceOrderComponent;
+  let fixture: ComponentFixture<PlaceOrderComponent>;
+  let controls: FormGroup['controls'];
+  let checkoutService: CheckoutFacade;
+  let routingService: RoutingService;
+  let launchDialogService: LaunchDialogService;
+
+  beforeEach(
+    waitForAsync(() => {
+      TestBed.configureTestingModule({
+        imports: [ReactiveFormsModule, RouterTestingModule, I18nTestingModule],
+        declarations: [MockUrlPipe, PlaceOrderComponent],
+        providers: [
+          { provide: CheckoutFacade, useClass: MockCheckoutService },
+          { provide: RoutingService, useClass: MockRoutingService },
+          { provide: LaunchDialogService, useClass: MockLaunchDialogService },
+        ],
+      }).compileComponents();
+    })
+  );
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(PlaceOrderComponent);
+    component = fixture.componentInstance;
+    controls = component.checkoutSubmitForm.controls;
+
+    checkoutService = TestBed.inject(CheckoutFacade);
+    routingService = TestBed.inject(RoutingService);
+    launchDialogService = TestBed.inject(LaunchDialogService);
+
+    spyOn(checkoutService, 'placeOrder').and.callThrough();
+  });
+
+  it('should be created', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should not place order when checkbox not checked', () => {
+    submitForm(false);
+
+    expect(checkoutService.placeOrder).not.toHaveBeenCalled();
+  });
+
+  it('should place order when checkbox checked', () => {
+    submitForm(true);
+
+    expect(checkoutService.placeOrder).toHaveBeenCalled();
+  });
+
+  it('should change page and reset form data on a successful place order', () => {
+    spyOn(routingService, 'go').and.stub();
+
+    component.onSuccess();
+
+    expect(routingService.go).toHaveBeenCalledWith({
+      cxRoute: 'orderConfirmation',
+    });
+  });
+
+  describe('when order was successfully placed', () => {
+    beforeEach(() => {
+      controls.termsAndConditions.setValue(true);
+    });
+    it('should open popover dialog', () => {
+      spyOn(launchDialogService, 'launch').and.stub();
+
+      component.submitForm();
+
+      expect(launchDialogService.launch).toHaveBeenCalledWith(
+        LAUNCH_CALLER.PLACE_ORDER_SPINNER,
+        component['vcr']
+      );
+    });
+  });
+
+  describe('Place order UI', () => {
+    beforeEach(() => {
+      controls.termsAndConditions.setValue(true);
+    });
+
+    it('should have button ENABLED when a checkbox for weekday in WEEKLY view is checked and terms and condition checked', () => {
```

Comment:
Weekly is gone from the test as that was for b2b scheduled replenishment

Let's bring back the DISABLED assertion in this UI test, so when the termsAndCondition is false

```suggestion
      it('should have the place order button ENABLED when terms and condition is checked', () => {
```

Reply:
Hmm, not sure how much sense is there to test this case?
If you look at the code, this is what we have:

```ts
if (this.checkoutSubmitForm.valid) {
      ...
      this.checkoutService.placeOrder(this.checkoutSubmitForm.valid).subscribe({
...})
```

which means that we will place the order only if the form is valid, and we pass the same flag as the `termsChecked` flag (kind of weird, but that's how it was implemented initially).
so I'm not sure how we can test it.

Reply:
true